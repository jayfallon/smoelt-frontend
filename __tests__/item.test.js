import ItemComponent from "../components/Item";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

const fakeItem = {
	id: "ABC123",
	title: "A basic item",
	price: 9998,
	description: "An item to test with",
	image: "item.jpg",
	largeImage: "largeItem.jpg",
};

describe("<Item/>", () => {
	it("renders and matches the snapshot", () => {
		const wrapper = shallow(<ItemComponent item={fakeItem} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
	// it("renders the priceTag and title properly", () => {
	// 	const wrapper = shallow(<ItemComponent item={fakeItem} />);
	// 	const PriceTag = wrapper.find("PriceTag");
	// 	// console.log(PriceTag.dive().text());
	// 	expect(PriceTag.children().text()).toBe("$99.99");
	// 	expect(wrapper.find("Title a").text()).toBe(fakeItem.title);
	// 	console.log(wrapper.debug());
	// });

	// it("renders the image properly", () => {
	// 	const wrapper = shallow(<ItemComponent item={fakeItem} />);
	// 	const img = wrapper.find("img");
	// 	// console.log(img.props());
	// 	expect(img.props().src).toBe(fakeItem.image);
	// 	expect(img.props().alt).toBe(fakeItem.title);
	// });

	// it("renders out the buttons properly", () => {
	// 	const wrapper = shallow(<ItemComponent item={fakeItem} />);
	// 	const buttonList = wrapper.find(".buttonList");
	// 	expect(buttonList.children()).toHaveLength(3);
	// 	expect(buttonList.find("Link").exists()).toBeTruthy();
	// 	expect(buttonList.find("AddToCart").exists()).toBeTruthy();
	// 	expect(buttonList.find("DeletItem").exists()).toBeTruthy();
	// });
});
