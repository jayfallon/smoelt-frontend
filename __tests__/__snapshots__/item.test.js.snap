// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A basic item"
    src="item.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A basic item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $99.98
  </PriceTag>
  <p>
    An item to test with
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeletItem
      id="ABC123"
    >
      Delete This Item
    </DeletItem>
  </div>
</ItemStyles__Item>
`;
